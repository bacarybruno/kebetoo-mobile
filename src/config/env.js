/* eslint-disable no-undef */

__DEV__ = false

// TODO: use env variables
export default {
  apiBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.herokuapp.com',
  assetsBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.s3.eu-west-3.amazonaws.com',
}
