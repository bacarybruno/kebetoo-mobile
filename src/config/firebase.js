import { utils } from '@react-native-firebase/app';
import analytics from '@react-native-firebase/analytics';

if (utils().isRunningInTestLab) {
  // do not track analytics for firebase test lab
  // used by google play to run test on publish
  analytics().setAnalyticsCollectionEnabled(false);
}
