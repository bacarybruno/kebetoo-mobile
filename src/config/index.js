export { default as env } from './env';
export { default as strings } from './strings';
