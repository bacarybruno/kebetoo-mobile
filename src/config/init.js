import { GoogleSignin } from '@react-native-community/google-signin'
import dayjs from 'dayjs'
import relativeTime from 'dayjs/plugin/relativeTime'

GoogleSignin.configure({
  scopes: ['email', 'profile', 'openid'],
  webClientId: '355888072499-spu5d874qrfb6tfs3pq195cjf6ij56go.apps.googleusercontent.com',
})

dayjs.extend(relativeTime)
