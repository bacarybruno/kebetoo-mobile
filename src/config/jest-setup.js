/* eslint-disable import/no-extraneous-dependencies */
import { NativeModules } from 'react-native';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MockDate from 'mockdate';
import configureStore from 'redux-mock-store';
import 'react-native-gesture-handler/jestSetup';
import { renderHook } from '@testing-library/react-hooks';
import auth from '@react-native-firebase/auth';
import { SafeAreaContext } from '@app/shared/contexts';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { PortalProvider } from '@gorhom/portal';

// Silence the warning https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

// wix/react-native-keyboard-input
NativeModules.KeyboardTrackingViewTempManager = {
  KeyboardTrackingScrollBehaviorNone: 'NONE',
  KeyboardTrackingScrollBehaviorScrollToBottomInvertedOnly: 'SCROLL_TO_BOTTOM_INVERTED_ONLY',
  KeyboardTrackingScrollBehaviorFixedOffset: 'FIXED_OFFSET',
};

NativeModules.RealPathUtils = {
  getOriginalFilePath: jest.fn(),
};

const mockedStoreState = {
  postsReducer: {
    posts: [],
    authors: [],
  },
  notificationsReducer: {
    notifications: [],
  },
  userReducer: {
    profile: {
      ...auth().currentUser,
      isLoggedIn: true,
    },
  },
  appReducer: {
    theme: 'system',
    locale: 'fr',
  },
};
// helper to setup unit tests
/* eslint-disable comma-dangle */
const setupTest = (WrappedComponent, renderFn = TestRenderer.create) => {
  let wrapper = null;
  return (defaultProps = {}) => (additionalProps = {}) => {
    const {
      store, __storeState__ = mockedStoreState, disconnectNavigation, ...props
    } = defaultProps;
    const propsWithArgs = {
      ...props,
      ...additionalProps,
    };
    const mockStore = configureStore();
    const safeAreaCtxValue = {
      updateTopSafeAreaColor: jest.fn(),
      updateBottomSafeAreaColor: jest.fn(),
      resetStatusBars: jest.fn(),
    };
    const Stack = createStackNavigator();
    const Component = (
      <SafeAreaContext.Provider value={safeAreaCtxValue}>
        <Provider store={store || mockStore(__storeState__)}>
          <PortalProvider>
            {disconnectNavigation
              ? <WrappedComponent {...propsWithArgs} />
              : (
                <NavigationContainer>
                  <Stack.Navigator>
                    <Stack.Screen name="jest">
                      {() => <WrappedComponent {...propsWithArgs} />}
                    </Stack.Screen>
                  </Stack.Navigator>
                </NavigationContainer>
              )}
          </PortalProvider>
        </Provider>
      </SafeAreaContext.Provider>
    );
    wrapper = renderFn(Component);
    return { wrapper, props: propsWithArgs };
  };
};

export const setupHook = (useHook, ...props) => {
  const mockStore = configureStore();
  const store = mockStore(mockedStoreState);
  const wrapper = ({ children }) => (
    <Provider store={store}>
      {children}
    </Provider>
  );
  const rendered = renderHook(() => useHook(...props), {
    wrapper,
  });
  return rendered;
};

/**
 * Explicit mocks
 * Create mocks here only if they can't be created on the __mocks__ folder
 */

jest.useFakeTimers();
jest.setTimeout(30000);

// react-navigation
const mockedNavigate = jest.fn();
const mockedSetOptions = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: mockedNavigate,
    setOptions: mockedSetOptions,
    addListener: jest.fn(),
  }),
  useRoute: () => ({
    params: {},
  }),
  useFocusEffect: (cb) => cb()
}));
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => { };
  Reanimated.default.addWhitelistedUIProps = () => { };

  return Reanimated;
});

// react-native-screens
jest.mock('react-native-screens', () => {
  const Screens = jest.requireActual('react-native-screens');
  Screens.enableScreens = () => { };
  return Screens;
});

// date
MockDate.set(new Date(Date.parse('2020-06-25T13:30:00+02:00')));

export default setupTest;
