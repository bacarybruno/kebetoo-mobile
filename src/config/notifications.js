import { Platform } from 'react-native'
import PushNotification from 'react-native-push-notification'
import AsyncStorage from '@react-native-community/async-storage'
import messaging from '@react-native-firebase/messaging'

import { getNotificationMessage, getNotificationTitle } from '@app/features/notifications/containers'

PushNotification.createChannel({
  channelId: 'kbt',
  channelName: 'Kebetoo Channel',
  channelDescription: 'Local notifications sent by Kebetoo',
  playSound: false,
})

const formatNotificationMessage = ({ name, message }) => `${name} ${message}`

PushNotification.configure({
  requestPermissions: Platform.OS === 'ios',
})

// handle background notifications
messaging().setBackgroundMessageHandler(async (remoteMessage) => {
  const bgNotifications = JSON.parse(await AsyncStorage.getItem('backgroundNotifications')) || []
  bgNotifications.push(remoteMessage)
  PushNotification.setApplicationIconBadgeNumber(bgNotifications.length)
  await AsyncStorage.setItem('backgroundNotifications', JSON.stringify(bgNotifications))

  const notificationTitle = getNotificationTitle(remoteMessage)
  if (!notificationTitle && !remoteMessage.data.title) return

  PushNotification.localNotification({
    channelId: 'kbt',
    title: notificationTitle
      ? formatNotificationMessage(notificationTitle)
      : remoteMessage.data.title,
    message: getNotificationMessage(remoteMessage) || remoteMessage.data.message,
    smallIcon: 'ic_stat_ic_notification',
  })
})
