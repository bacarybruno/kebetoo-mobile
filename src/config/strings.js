import LocalizedStrings from 'react-native-localization';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import 'dayjs/locale/en';
import 'dayjs/locale/fr';

import frStrings from '@assets/locales/fr';
import enStrings from '@assets/locales/en';

const strings = new LocalizedStrings({
  fr: frStrings,
  en: enStrings,
});

dayjs.locale(strings.getLanguage());
dayjs.extend(relativeTime);

export const updateAppLocale = (locale) => {
  dayjs.locale(locale);
  strings.setLanguage(locale);
};

export default strings;
