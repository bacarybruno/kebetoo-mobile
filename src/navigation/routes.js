const routes = {
  ONBARDING_NAV: 'route:nav:onboarding',
  SIGNIN: 'route:signin',
  SIGNUP: 'route:signup',
  ONBOARDING: 'route:obnoarding',
  HOME_NAV: 'route:nav:home',
  HOME: 'route:home',
  PROFILE: 'route:profile',
  SEARCH: 'route:search',
  STORIES: 'route:stories',
  CREATE_POST: 'route:create_post',
  TABS_FAB: 'route:custom:tabs_fab',
}

export default routes
