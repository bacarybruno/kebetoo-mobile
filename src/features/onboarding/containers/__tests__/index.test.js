import { render, fireEvent } from 'react-native-testing-library';

import setupTest from '@app/config/jest-setup';

import { strings } from '@app/config';
import routes from '@app/navigation/routes';

import Onboarding, { slideItems } from '../index';

afterEach(jest.clearAllMocks);

const givenOnboarding = setupTest(Onboarding, render)({
  navigation: {
    setOptions: jest.fn(),
    navigate: jest.fn(),
    addListener: jest.fn().mockImplementation((e, callback) => { callback(); return jest.fn(); }),
  },
});

it('renders Onboarding', () => {
  const { wrapper } = givenOnboarding();
  expect(wrapper.toJSON()).toMatchSnapshot();
});

it('hides navigation header by default', () => {
  const { props } = givenOnboarding();
  expect(props.navigation.setOptions).toBeCalledWith({ headerShown: false });
});

describe('skip button', () => {
  it('shows the button by default', () => {
    const { wrapper } = givenOnboarding();
    expect(wrapper.getAllByText(strings.general.skip).length).toBe(1);
  });

  it('goes to signup if user skips the onboarding', () => {
    const { wrapper, props } = givenOnboarding();
    const skipButton = wrapper.getByText(strings.general.skip);
    fireEvent.press(skipButton);
    expect(props.navigation.navigate).toBeCalledTimes(1);
    expect(props.navigation.navigate).toBeCalledWith(routes.SIGNUP);
  });

  it('hides the button on last slide item', () => {
    const { wrapper } = givenOnboarding();
    const nextButton = wrapper.getByTestId('next-button');
    const swiper = wrapper.getByTestId('swiper');
    for (let i = 0; i < slideItems.length; i += 1) {
      fireEvent.press(nextButton);
      // this event should be fired automatically
      // but onLayout is not being called on unit tests
      fireEvent(swiper, 'onIndexChanged', i);
      if (i < slideItems.length - 1) {
        expect(wrapper.queryByText(strings.general.skip)).not.toBeNull();
      }
    }
    expect(wrapper.queryByText(strings.general.skip)).toBeNull();
  });
});

describe('submit button', () => {
  it('shows the button only on last slide items', () => {
    const { wrapper } = givenOnboarding();
    const nextButton = wrapper.getByTestId('next-button');
    const swiper = wrapper.getByTestId('swiper');
    for (let i = 0; i < slideItems.length; i += 1) {
      fireEvent.press(nextButton);
      // this event should be fired automatically
      // but onLayout is not being called on unit tests
      fireEvent(swiper, 'onIndexChanged', i);
      if (i < slideItems.length - 1) {
        expect(wrapper.queryByText(strings.general.get_started.toUpperCase())).toBeNull();
      }
    }
    expect(wrapper.queryByText(strings.general.get_started.toUpperCase())).not.toBeNull();
  });

  it('navigates to signup on submit', () => {
    const { wrapper, props } = givenOnboarding();
    const swiper = wrapper.getByTestId('swiper');
    fireEvent(swiper, 'onIndexChanged', slideItems.length - 1);
    fireEvent.press(wrapper.getByText(strings.general.get_started.toUpperCase()));
    expect(props.navigation.navigate).toBeCalledTimes(1);
    expect(props.navigation.navigate).toBeCalledWith(routes.SIGNUP);
  });
});
