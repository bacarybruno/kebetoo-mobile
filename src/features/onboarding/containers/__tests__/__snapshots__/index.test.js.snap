// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Onboarding 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#E0F2F8",
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/onboarding1.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginVertical": 24,
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "flex-end",
            "marginHorizontal": 16,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": -0.32,
                "lineHeight": 21,
              },
              Array [
                Object {
                  "color": "rgba(0,0,0,0.54)",
                },
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                undefined,
              ],
            ]
          }
        >
          Skip
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <RCTScrollView
        activeDotStyle={
          Object {
            "backgroundColor": "rgba(0,0,0,0.87)",
            "height": 8,
            "width": 15,
          }
        }
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDirection={true}
        autoplayTimeout={2.5}
        bounces={false}
        contentContainerStyle={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginTop": 11,
            },
          ]
        }
        contentOffset={
          Object {
            "x": 0,
          }
        }
        disableNextButton={false}
        disablePrevButton={false}
        dotStyle={
          Object {
            "backgroundColor": "rgba(199, 199, 204, 255)",
            "height": 8,
            "width": 8,
          }
        }
        horizontal={true}
        index={0}
        loadMinimal={false}
        loadMinimalSize={1}
        loop={true}
        onIndexChanged={[Function]}
        onMomentumScrollEnd={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        paginationStyle={
          Object {
            "justifyContent": "flex-start",
            "marginHorizontal": 16,
          }
        }
        pagingEnabled={true}
        removeClippedSubviews={true}
        scrollEnabled={false}
        scrollsToTop={false}
        showsButtons={false}
        showsHorizontalScrollIndicator={false}
        showsPagination={true}
        showsVerticalScrollIndicator={false}
        testID="swiper"
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "height": 1334,
                  "width": 750,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 733.7,
                }
              }
            >
              <Text
                bold={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 28,
                      "fontWeight": "400",
                      "letterSpacing": 0.355469,
                      "lineHeight": 34,
                    },
                    false,
                    false,
                    false,
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "600",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                React to the Contents
You Care About
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "400",
                      "letterSpacing": -0.408,
                      "lineHeight": 22,
                    },
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "marginTop": 46.690000000000005,
                      },
                    ],
                  ]
                }
              >
                React and comment on the posts that interest you the most in a simple and intuitive way ‚ù§Ô∏è
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 1334,
                  "width": 750,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 733.7,
                }
              }
            >
              <Text
                bold={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 28,
                      "fontWeight": "400",
                      "letterSpacing": 0.355469,
                      "lineHeight": 34,
                    },
                    false,
                    false,
                    false,
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "600",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Join Our Social Media
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "400",
                      "letterSpacing": -0.408,
                      "lineHeight": 22,
                    },
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "marginTop": 46.690000000000005,
                      },
                    ],
                  ]
                }
              >
                Kebetoo is a microblogging app with African DNA. Be one of the first to join the adventure üí™
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 1334,
                  "width": 750,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 733.7,
                }
              }
            >
              <Text
                bold={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 28,
                      "fontWeight": "400",
                      "letterSpacing": 0.355469,
                      "lineHeight": 34,
                    },
                    false,
                    false,
                    false,
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "600",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                It's your turn to speak
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "400",
                      "letterSpacing": -0.408,
                      "lineHeight": 22,
                    },
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "marginTop": 46.690000000000005,
                      },
                    ],
                  ]
                }
              >
                With Kebetoo, you can create and share public posts of all types: videos, images, texts but also... voice messages üé§
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 1334,
                  "width": 750,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 733.7,
                }
              }
            >
              <Text
                bold={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 28,
                      "fontWeight": "400",
                      "letterSpacing": 0.355469,
                      "lineHeight": 34,
                    },
                    false,
                    false,
                    false,
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "600",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                React to the Contents
You Care About
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "400",
                      "letterSpacing": -0.408,
                      "lineHeight": 22,
                    },
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "marginTop": 46.690000000000005,
                      },
                    ],
                  ]
                }
              >
                React and comment on the posts that interest you the most in a simple and intuitive way ‚ù§Ô∏è
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 1334,
                  "width": 750,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 733.7,
                }
              }
            >
              <Text
                bold={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 28,
                      "fontWeight": "400",
                      "letterSpacing": 0.355469,
                      "lineHeight": 34,
                    },
                    false,
                    false,
                    false,
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "600",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Join Our Social Media
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "400",
                      "letterSpacing": -0.408,
                      "lineHeight": 22,
                    },
                    Array [
                      Object {
                        "color": "rgba(0,0,0,0.87)",
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "marginTop": 46.690000000000005,
                      },
                    ],
                  ]
                }
              >
                Kebetoo is a microblogging app with African DNA. Be one of the first to join the adventure üí™
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "bottom": 25,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
            },
            Object {
              "justifyContent": "flex-start",
              "marginHorizontal": 16,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#007aff",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 3,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 3,
                "width": 8,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0.87)",
                "height": 8,
                "width": 15,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 3,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 3,
                "width": 8,
              },
              Object {
                "backgroundColor": "rgba(199, 199, 204, 255)",
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 3,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 3,
                "width": 8,
              },
              Object {
                "backgroundColor": "rgba(199, 199, 204, 255)",
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 8,
          "position": "absolute",
          "right": 16,
          "width": 718,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 122, 255, 255)",
            "borderRadius": 15,
            "bottom": 0,
            "color": "rgba(255, 255, 255, 255)",
            "elevation": 2,
            "height": 48,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "rgba(0, 0, 0, 255)",
            "shadowOffset": Object {
              "x": 0,
              "y": 1,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 1.41,
            "width": 57,
          }
        }
        testID="next-button"
      >
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "rgba(255, 255, 255, 255)",
              "fontFamily": "Ionicons",
              "fontSize": 30,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
          text="ionicon-ios-arrow-round-forward"
        />
      </View>
    </View>
  </View>
</View>
`;
